orttraining/orttraining/training_api/optimizer.h

```diff
@@ -66,9 +102,25 @@ struct Optimizer {
 
   Status Step();
 
+  /**
+   * @brief Get the current optimizer state.
+   *
+   * Be noted the returned optimizer_checkpoint_states will hold new references to
+   * original momentum states.
+   * @return Status
+   */
   Status GetStateDict(OptimizerCheckpointState& optimizer_checkpoint_states);
```

Comment:
Do we still need this?

Reply:
Yes, this is needed for saving checkpoint.

Reply:
This should no longer be needed. Users own the `CheckpointState` and the optimizer directly accesses/manipulates this checkpoint state's optimizer_state member.

The expectation is that at any given point, we should not need to make a copy of the state (other than moving data between devices).

Reply:
OK, You guys are right, I just realized our C/C++API samples have removing usage of it completely. I planned to remove it later since some UTs are still using it, I may defer the change a bit. 

 